packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,597 @@
+            repo_array = [x for x in self.repo_list if x['id'] == repo_id]
+            self.repo_cache[repo_id] = repo_array[0]
+
+        return self.repo_cache[repo_id]
+
+    def publish_repo(self, repo_id, publish_distributor):
+        url = "%s/pulp/api/v2/repositories/%s/actions/publish/" % (self.host, repo_id)
+
+        # If there's no distributor specified, we will publish them all
+        if publish_distributor is None:
+            repo_config = self.get_repo_config_by_id(repo_id)
+
+            for distributor in repo_config['distributors']:
+                data = dict()
+                data['id'] = distributor['id']
+                response, info = fetch_url(
+                    self.module,
+                    url,
+                    data=json.dumps(data),
+                    method='POST')
+
+                if info['status'] != 202:
+                    self.module.fail_json(
+                        msg="Failed to publish the repo the repo.",
+                        status_code=info['status'],
+                        response=info['msg'],
+                        url=url,
+                        distributor=distributor['id']
+                    )
+        else:
+            data = dict()
+            data['id'] = publish_distributor
+            response, info = fetch_url(
+                self.module,
+                url,
+                data=json.dumps(data),
+                method='POST')
+            if info['status'] != 202:
+                self.module.fail_json(
+                    msg="Failed to schedule a sync of the repo.",
+                    status_code=info['status'],
+                    response=info['msg'],
+                    url=url,
+                    distributor=publish_distributor
+                )
+        return True
+
+    def sync_repo(self, repo_id):
+        url = "%s/pulp/api/v2/repositories/%s/actions/sync/" % (self.host, repo_id)
+        response, info = fetch_url(self.module, url, data='', method='POST')
+
+        if info['status'] != 202:
+            self.module.fail_json(
+                msg="Failed to schedule a sync of the repo.",
+                status_code=info['status'],
+                response=info['msg'],
+                url=url)
+        return True
+
+    def update_repo_distributor_config(self, repo_id, **kwargs):
+        url = "%s/pulp/api/v2/repositories/%s/distributors/" % (self.host, repo_id)
+        repo_config = self.get_repo_config_by_id(repo_id)
+
+        for distributor in repo_config['distributors']:
+            distributor_url = "%s%s/" % (url, distributor['id'])
+            data = dict()
+            data['distributor_config'] = dict()
+
+            for key, value in kwargs.iteritems():
+                data['distributor_config'][key] = value
+
+            response, info = fetch_url(
+                self.module,
+                distributor_url,
+                data=json.dumps(data),
+                method='PUT'
+            )
```

Comment:
I would consider to move the ending parenthesis to the last like of code to keep it the same like in the case of `self.module.fail_json(...)` or `fetch_url(...)`.

```
response, info = fetch_url(
  self.module,
  distributor_url,
  data=json.dumps(data),
  method='PUT')
```

There is several other occurrences elsewhere in the code.